### PR TITLE
Introduce dagger + minor fix to snackbar

### PR DESCRIPTION
**Pull request by worker8**
### Overview
Use Dagger 2 Android for dependency injection.

### CODE REVIEW

**Comment by worker8-1** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          13 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/build.gradle#L43">app/build.gradle#L43</a> - A newer version of androidx.appcompat:appcompat than 1.1.0-alpha01 is available: 1.1.0-alpha02</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/build.gradle#L44">app/build.gradle#L44</a> - A newer version of androidx.recyclerview:recyclerview than 1.1.0-alpha01 is available: 1.1.0-alpha02</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/build.gradle#L46">app/build.gradle#L46</a> - A newer version of androidx.paging:paging-runtime than 2.1.0-rc01 is available: 2.1.0</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/build.gradle#L60">app/build.gradle#L60</a> - A newer version of androidx.lifecycle:lifecycle-extensions than 2.1.0-alpha01 is available: 2.1.0-alpha02</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/src/main/res/drawable/ic_logo_foreground.xml#L15">app/src/main/res/drawable/ic_logo_foreground.xml#L15</a> - Attribute <code>endX</code> is only used in API level 24 and higher (current min is 21)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/src/main/res/drawable/ic_logo_foreground.xml#L16">app/src/main/res/drawable/ic_logo_foreground.xml#L16</a> - Attribute <code>endY</code> is only used in API level 24 and higher (current min is 21)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/src/main/res/drawable/ic_logo_foreground.xml#L17">app/src/main/res/drawable/ic_logo_foreground.xml#L17</a> - Attribute <code>startX</code> is only used in API level 24 and higher (current min is 21)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/src/main/res/drawable/ic_logo_foreground.xml#L18">app/src/main/res/drawable/ic_logo_foreground.xml#L18</a> - Attribute <code>startY</code> is only used in API level 24 and higher (current min is 21)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/src/main/res/drawable/ic_logo_foreground.xml#L22">app/src/main/res/drawable/ic_logo_foreground.xml#L22</a> - Attribute <code>offset</code> is only used in API level 24 and higher (current min is 21)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/src/main/res/drawable/ic_logo_foreground.xml#L25">app/src/main/res/drawable/ic_logo_foreground.xml#L25</a> - Attribute <code>offset</code> is only used in API level 24 and higher (current min is 21)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/src/main/res/drawable/ic_outline_bug.xml#L8">app/src/main/res/drawable/ic_outline_bug.xml#L8</a> - Very long vector path (852 characters), which is bad for performance. Considering reducing precision, removing minor details or rasterizing vector.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/src/main/res/layout/comment_header.xml#L7">app/src/main/res/layout/comment_header.xml#L7</a> - Possible overdraw: Root element paints background <code>?pixelBackgroundColor</code> with a theme that also paints a background (inferred theme is <code>@style/AppTheme</code>)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/worker8/Pixels/blob/570c219bac76fae5c87718687ccdba7332e3efce/app/src/main/res/mipmap-hdpi/ic_launcher_round.png#L0">app/src/main/res/mipmap-hdpi/ic_launcher_round.png#L0</a> - Launcher icon used as round icon did not have a circular shape</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/main/java/beepbeep/pixelsforredditx/common/SnackbarOnlyOne.kt [hunk 1]
```diff
@@ -24,11 +27,19 @@ class SnackbarOnlyOne() {
                     }
                 })
                 snackbar = this
+                val backgroundColor = getThemeBackgroundColor(this@apply.view.context)
```
Comment (kittinunf) [inline app/src/main/java/beepbeep/pixelsforredditx/common/SnackbarOnlyOne.kt]: Why sudden we need this in this PR bro?

File: app/src/main/java/beepbeep/pixelsforredditx/ui/home/HomeActivity.kt [hunk 2]
```diff
@@ -115,17 +121,17 @@ class HomeActivity : AppCompatActivity() {
...
+            setTheme(beepbeep.pixelsforredditx.R.style.AppTheme)
         }
     }
 
     private fun setupView() {
         selectedSubredditToolbar.text = RedditPreference.getSelectedSubreddit(this)
         adapter.apply {
             homeList.adapter = this
-            homeList.addItemDecoration(DividerItemDecoration(homeList.context, DividerItemDecoration.VERTICAL).apply { setDrawable(resources.getDrawable(R.drawable.recycler_view_divider)) })
+            homeList.addItemDecoration(DividerItemDecoration(homeList.context, DividerItemDecoration.VERTICAL).apply { setDrawable(resources.getDrawable(beepbeep.pixelsforredditx.R.drawable.recycler_view_divider)) })
```
Comment (kittinunf) [inline app/src/main/java/beepbeep/pixelsforredditx/ui/home/HomeActivity.kt]: full qualified name R?

File: app/src/main/java/beepbeep/pixelsforredditx/ui/home/di/HomeModule.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package beepbeep.pixelsforredditx.ui.home.di
+
+import beepbeep.pixelsforredditx.PixelApplication
+import beepbeep.pixelsforredditx.ui.home.HomeRepo
+import dagger.Module
+import dagger.Provides
+
+@Module
+class HomeModule() {
```
Comment (kittinunf) [inline app/src/main/java/beepbeep/pixelsforredditx/ui/home/di/HomeModule.kt]: We need this empty file right?

File: buildSrc/src/main/java/Versions.kt [hunk 4]
```diff
@@ -26,6 +26,7 @@ object Versions {
     val result = "2.0.0"
     val fuel = "1.15.1"
     val mockk = "1.8.13.kotlin13"
+    val dagger = "2.21"
```
Comment (kittinunf) [inline buildSrc/src/main/java/Versions.kt]: Sort them?

**Comment by kittinunf** [comment]
💡 I think you should rebase on top of updated master.

File: app/src/main/java/beepbeep/pixelsforredditx/common/SnackbarOnlyOne.kt [hunk 5]
```diff
@@ -24,11 +27,19 @@ class SnackbarOnlyOne() {
...
             .show()
     }
 
     fun dismiss() {
         snackbar?.dismiss()
     }
+
+    fun getThemeBackgroundColor(context: Context): Int {
+        val value = TypedValue()
+        context.getTheme().resolveAttribute(beepbeep.pixelsforredditx.R.attr.background, value, true)
```
Comment (kittinunf) [inline app/src/main/java/beepbeep/pixelsforredditx/common/SnackbarOnlyOne.kt]: FULL QUALIFIED `R`

Comment (kittinunf) [inline app/src/main/java/beepbeep/pixelsforredditx/common/SnackbarOnlyOne.kt]:
```suggestion
        context.getTheme().resolveAttribute(R.attr.background, value, true)
```

File: app/src/main/java/beepbeep/pixelsforredditx/common/SnackbarOnlyOne.kt (hunk 1 above)
Comment (worker8) [reply app/src/main/java/beepbeep/pixelsforredditx/common/SnackbarOnlyOne.kt]: ah this is to fix another bug where the text is not visible in Snackbar in dark theme due to wrong color

**Pull request merged**